x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/AggregateExec.java

```diff
@@ -184,6 +187,25 @@ protected AttributeSet computeReferences() {
         return mode.isInputPartial() ? new AttributeSet(intermediateAttributes) : Aggregate.computeReferences(aggregates, groupings);
     }
 
+    /** Returns the attributes that can be loaded from ordinals -- no explicit extraction is needed */
+    public List<Attribute> ordinalAttributes() {
+        List<Attribute> orginalAttributs = new ArrayList<>(groupings.size());
+        if (groupings().size() == 1) {
+            // CATEGORIZE requires the standard hash aggregator as well.
+            if (groupings.get(0).anyMatch(e -> e instanceof Categorize) == false) {
```

Comment:
```suggestion
            if (groupings.get(0).noneMatch(e -> e instanceof Categorize)) {
```

I think this could also be merged with above `if` to minimize nesting.

Reply:
`groupings` is a list of `Expression` (not a stream), so won't have `noneMatch()`. But I think refuting a predicate is quite common, adding a `noneMatch()` would actually make sense. Will leave this as is for now, but will follow up.

> I think this could also be merged with above if to minimize nesting.

I've only grafted this patch, so I can only guess that the intention might have been to allow for more exceptions (or easier comments). But maybe this can be now merged and split again when needed.